File: lecture011/maxi.cpp
#include<iostream>
#include<vector>
#include<climits>

using namespace std ;
int maximum(vector<int> &arr , int idx){
    if(idx==arr.size())
    return INT_MIN;

    if(idx==arr.size()-1)
    return arr[idx];

    int recMax=maximum(arr,idx+1);

    //return max(arr[idx],recMax);

    if(arr[idx]>recMax){
         recMax=arr[idx];
        }
        return recMax;
}
int main(int args , char** argv){
    vector<int> arr={1,2,3,4,5,6,77,89,9,0,-2};
    cout<<maximum(arr,0);
}

File: lecture011/maxi.java
import java.util.Scanner;

public class maxi{
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args){
        int [] arr = {1,2,3,4,5,6,7,8,75,8,9,68,9,9,8,6,8};
        System.out.println(maxi(arr, 0)); 
    }
    public static int maxi(int [] arr , int idx){
        // if(idx==arr.length)
        // return INT_MIN;
        
        if(idx==arr.length-1)
        return arr[idx];

        int ans=maxi(arr , idx+1);
        
        if(arr[idx]>ans)
        ans=arr[idx];

        return ans;
    }
}